File: c/21.c
#include <stdio.h>

int divsum(int n)
{
	int i, sum = 0;
	for (i = 1; i <= n / 2; ++i)
		sum += n % i ? 0 : i;
	return sum;
}

int main(void)
{
	int i, ds, sum = 0;

	for (i = 2; i < 10000; ++i) {
		ds = divsum(i);
		if (i == divsum(ds) && i != ds)
			sum += i;
	}

	printf("%d\n", sum);
	return 0;
}

File: py/21.py
def divsum(n):
    return reduce(lambda a, b: a + b,
           filter(lambda m: n % m == 0, xrange(1, n / 2 + 1)), 0)

sum = 0

for i in xrange(2, 10001):
    ds = divsum(i)
    if i == divsum(ds) and i != ds:
        sum += i

print sum
